File: Automation/001-AutomationFramework/C#/AutomationFramework/Common/CommonCore/Tests/IStepInfo.cs
namespace CommonCore.Tests
{
    public interface IStepInfo
    {
        string Description { get; set; }
        bool SkipStep { get; set; }
        Priority Level { get; set; }
        bool SkipStepOnFailure { get; set; }
        bool FailsIteration { get; set; }
        TestState Status { get; set; }
        DateTime StartTime { get; set; }
        DateTime EndTime { get; set; }
        TimeSpan Duration { get; set; }
        Exception StepException { get; set; }
        bool IsMandatory { get; set; }
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/main/java/common/core/contracts/IStepInfo.java
package common.core.contracts;

import common.core.tests.Priority;
import common.core.tests.TestState;

import java.time.Duration;
import java.time.LocalDateTime;

public interface IStepInfo {
    String description = "";
    boolean skipStep = false;
    Priority level = null;
    boolean skipStepOnFailure = false;
    boolean failsIteration = false;
    TestState status = null;
    LocalDateTime startTime = null;
    LocalDateTime endTime = null;
    Duration duration = null;
    Exception stepException = null;
    boolean isMandatory = false;
}
